projects/core/src/lib/operators/dialogs/time-shift/time-shift.component.ts

```diff
@@ -0,0 +1,214 @@
+import {Layer, RasterLayer, VectorLayer} from '../../../layers/layer.model';
+import {ResultTypes} from '../../result-type.model';
+import {AfterViewInit, ChangeDetectionStrategy, Component, OnDestroy, OnInit} from '@angular/core';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import {ProjectService} from '../../../project/project.service';
+import {geoengineValidators} from '../../../util/form.validators';
+import {map, mergeMap} from 'rxjs/operators';
+import {NotificationService} from '../../../notification.service';
+import {TimeStepGranularityDict, WorkflowDict} from '../../../backend/backend.model';
+import {BehaviorSubject, Observable} from 'rxjs';
+import {AbsoluteTimeShiftDictParams, RelativeTimeShiftDictParams, TimeShiftDict} from '../../../backend/operator.model';
+import moment from 'moment';
+import {RasterSymbology, VectorSymbology} from '../../../layers/symbology/symbology.model';
+import {Time} from '../../../time/time.model';
+
+type TimeShiftFormType = 'relative' | 'absolute';
+
+interface TimeShiftForm {
+    name: FormControl<string>;
+    source: FormControl<Layer | undefined>;
+    type: FormControl<TimeShiftFormType>;
+}
+
+interface AbsoluteTimeShiftForm extends TimeShiftForm {
+    type: FormControl<'absolute'>;
+    timeIntervalStart: FormControl<moment.Moment>;
+    timeIntervalEnd: FormControl<moment.Moment>;
+}
+
+interface RelativeTimeShiftForm extends TimeShiftForm {
+    type: FormControl<'relative'>;
+    granularity: FormControl<TimeStepGranularityDict>;
+    value: FormControl<number>;
+}
+
+@Component({
+    selector: 'geoengine-time-shift',
+    templateUrl: './time-shift.component.html',
+    styleUrls: ['./time-shift.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TimeShiftComponent implements OnInit, AfterViewInit, OnDestroy {
+    readonly inputTypes = [ResultTypes.RASTER, ...ResultTypes.VECTOR_TYPES];
+
+    readonly timeGranularityOptions: Array<TimeStepGranularityDict> = ['millis', 'seconds', 'minutes', 'hours', 'days', 'months', 'years'];
```

Comment:
as this is the third time we define this list, maybe we can define it at a single place?